proof-defns/proof_defns.tex

```diff
@@ -0,0 +1,438 @@
+
+\tableofcontents
+
+\section*{List of terms that have not yet been added}
+\begin{itemize}
+\item Generalization of the path property and the corresponding proofs
+\item Better definition for multisets, and an explanation of the bijection between multisets and histograms
+\item Clarification sentence on the stability relation
+\item Approach with floating point
+\end{itemize}
+
+\section{Mathematical operators}
+\questionc{We would like to discuss how to approach this section further. We are not clear on the relation between MPFR and functions like \texttt{saturating\_add}.}
+
+The notation we are using for various mathematical operations is, as of 24 June 2021, inspired by section 2 of \url{https://hal-ens-lyon.archives-ouvertes.fr/ensl-01529804/file/crlibm.pdf}. (The notation may change in the future.) We plan to use a similar standardized notation for describing the semantics of \href{https://www.mpfr.org/}{MPFR}.\footnote{Library used in OpenDP to deal with floating point arithmetic.}
+
+\begin{definition}[$+, -, \times$]
+    The symbols $+$, $-$, and $\times$ represent the usual mathematical operations of addition, subtraction, and multiplication, respectively.
+\end{definition}
+
+\begin{definition}[$\oplus, \ominus, \otimes$]
+    The symbols $\oplus$, $\ominus$, and $\otimes$ denote the corresponding floating point operations in \href{https://www.mpfr.org/}{MPFR} / in Rust arithmetic. 
+\end{definition}
+
+\connor{Maybe change ``floating point operations in \href{https://www.mpfr.org/}{MPFR}'' to some other thing that it denotes. I'm not sure if they should ``denote the corresponding floating point operations in \href{https://www.mpfr.org/}{MPFR}'', but I wanted to preserve the notation, and I wasn't sure what it should be defined on \ldots.}
+    
+\silvia{Use same notation in both pseudocode and proofs?}
+
+\todo{Change to cast.}\connor{I don't understand this todo; could someone else complete it?}
+\begin{definition}
+    Given a Rust value $x$, the function \texttt{val(x)} returns the actual (real or integer) number represented by the Rust value $x$.
+
+\end{definition}
+
+\horizline
+
+\subsection{Notes, todos, questions}
+
+\todo{Add the cast definition?}
+
+\todo{max, min, val(x) (give number represented by Rust value), see email from June 21. Subindex for type?}
+
+\todo{Add special float symbol for max, min?} No, from discussion on June 28: as long as we pass as arguments things that have used float symbols, it's OK.
+
+\questionc{Type in subindex vs casting type.}
+\connor{What does ``subindex for type'' mean (see todo above for use of this phrase; meaning possibly relates to question in this line)?} \silvia{I think it means deciding between, for example, $\texttt{max}_{\texttt{u32}}(x, y)$ or $\texttt{cast(max(x, y), u32)}$.}
+
+\section{Data Representation}
+
+\begin{definition}[Vector]
+A vector $v$ is an ordered list of objects.
```

Comment:
Maybe define a vector wrt a type, and/or stipulate that all members of the vector have the same type.

Reply:
Yes we are trying to keep the definitions for the proofs to only use mathematical and / or idealized terms, and use \texttt notation and Rust-related concepts in the document for definitions for the pseudocode.